diff --git a/fs/proc/base.c b/fs/proc/base.c
index 7235579..4a756a5 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -100,6 +100,10 @@
 #include <linux/cn_proc.h>
 #include <linux/ksm.h>
 #include <linux/cpufreq_times.h>
+#if defined(CONFIG_KSU_SUSFS_SUS_MAP) || defined(CONFIG_KSU_SUSFS_OPEN_REDIRECT)
+#include <linux/susfs_def.h>
+#endif // #if defined(CONFIG_KSU_SUSFS_SUS_MAP) || defined(CONFIG_KSU_SUSFS_OPEN_REDIRECT)
+
 #include <trace/events/oom.h>
 #include <trace/hooks/sched.h>
 #include "internal.h"
