diff --git a/fs/f2fs/compress.c b/fs/f2fs/compress.c
--- a/fs/f2fs/compress.c
+++ b/fs/f2fs/compress.c
@@ -290,9 +290,13 @@
 static int lz4_decompress_pages(struct decompress_io_ctx *dic)
 {
 	int ret;
-
+#if defined(CONFIG_ARM64) && defined(CONFIG_KERNEL_MODE_NEON)
+	ret = LZ4_arm64_decompress_safe(dic->cbuf->cdata, dic->rbuf,
+						dic->clen, dic->rlen, false);
+#else
 	ret = LZ4_decompress_safe(dic->cbuf->cdata, dic->rbuf,
-						dic->clen, dic->rlen);
+						dic->clen, dic->rlen, false);
+#endif
 	if (ret < 0) {
 		f2fs_err_ratelimited(F2FS_I_SB(dic->inode),
 				"lz4 decompress failed, ret:%d", ret);
